ex1/dates_times.py

```diff
@@ -33,12 +33,49 @@
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
 
+import datetime
+
+def last_of_month(dt):
+    """Return a date object with the last day of the month of the provided date
+    """
+    dt = dt.replace(day=1) + datetime.timedelta(days=31)
+    dt= dt.replace(day=1)
+    return dt - datetime.timedelta(days=1)
+    
+
 # Define a function named feed_the_gremlin which takes a time object as an
 # argument. It should return False between midnight and 6:30AM and True
 # otherwise.
 
+def feed_the_gremlin(time):
+    """Tells you wether or not to feed the gremlin at a given time
+    """
+    return not (time.hour <= 6)&(time.minute < 30)
+
+
 # Define a function named how_long that takes two datetime objects dt and ref
 # where ref is the reference datetime, calculates the difference between them and
 # returns the difference as a string formatted like:
 # "01 days, 01 minutes, 01 seconds until 2000-12-31 15:59:59"
 # If ref is before dt then use 'since' instead of 'until'
+
+dt = datetime.datetime(1987, 1, 31, 23, 50, 50)
+ref = datetime.datetime(1986, 3, 25, 21, 25, 14)
```

Comment:
I guess you forgot to delete those. Otherwise it is seldom a good idea to define global variables.